### PR TITLE
PPPSYS-39200 Add available consent types to developers documentation

### PR DESCRIPTION
**Pull request by archi-tektur**

### CODE REVIEW

**Comment by kororokke** [comment]
Master branch is the target?

**Comment by archi-tektur** [comment]
@kororokke If I understand this correctly, yes. It's not related to any changes in any release, just adding some explanation about exact values an API user can use.

**Review by musialp — COMMENTED** [review]
@archi-tektur I was wondering if maybe we could solve this in a similar manner as the custom dimensions object is handled [here](https://developers.piwik.pro/en/latest/data_collection/web/javascript_tracking_client/api.html#custom-events), so put the list in a separate section and link to it where needed. What do you think?

**Comment by archi-tektur** [comment]
@musialp You can check it out if it's working fine for you now :)

File: consent_manager/js_api/commands.rst [hunk 1]
```diff
@@ -368,3 +375,35 @@ Code::
     The rejection handler callback (called with error code). If not specified, the exception will be thrown in the main stack trace.
 
     :param string|object error: **Required** Error code or exception
+
+Compliance types reference
```
[the remark this answers is not in the document]
Comment (musialp) [reply consent_manager/js_api/commands.rst:379]: or
```suggestion
Available consent types
```

[the remark this answers is not in the document]
Comment (musialp) [reply consent_manager/js_api/commands.rst:379]: Compliance is only used in command names for compatibility reason. Whenever we refer to those particular values we call them `consent types`

File: consent_manager/js_api/commands.rst [hunk 2]
```diff
@@ -115,10 +124,8 @@ Code::
 
             {consents: {analytics: {status: 1}}}
 
-    Where ``consent.analytics`` is consent type and status indicate:
```
Comment (musialp) [inline consent_manager/js_api/commands.rst:118]: This was removed for `Set compliance settings` but stayed for `Get compliance settings`. IMO it should stay for both as the available status values are different.

File: consent_manager/js_api/commands.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (musialp) [reply consent_manager/js_api/commands.rst:379]: You mentioned that it is bothering you, that it looks like another command and I agree. What do you think about putting it as a top level section? Same as `Introduction` and `Commands`?

**Review by musialp — CHANGES_REQUESTED** [review]

File: consent_manager/js_api/commands.rst [hunk 3]
```diff
@@ -27,7 +27,10 @@ Code::
 
         Example::
 
-            ["remarketing", "analytics"]
+            ['remarketing', 'analytics']
+
+        .. note::
```
[the remark this answers is not in the document]
Comment (archi-tektur) [reply consent_manager/js_api/commands.rst:32]: fixed

File: consent_manager/js_api/commands.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (archi-tektur) [reply consent_manager/js_api/commands.rst:379]: Done just like you said it - looks good!

File: consent_manager/js_api/commands.rst [hunk 4]
```diff
@@ -368,3 +375,35 @@ Code::
...
+            marketing_automation: { status: 1 },
+            remarketing: { status: 1 },
+            conversion_tracking: { status: 1 }
+        ]
+    
+    Key ``consent.{consent_type}.status`` is indicating:
+
+        * ``-1`` - user has not interacted, e.g. has closed a consent popup without any decision
+        * ``0``  - user has rejected the consent
+        * ``1``  - user has approved the consent
```
[the remark this answers is not in the document]
Comment (archi-tektur) [reply consent_manager/js_api/commands.rst:409]: fixed

File: consent_manager/js_api/reference.rst [hunk 5]
```diff
@@ -0,0 +1,16 @@
+.. _`Piwik PRO - Custom consent form example`: https://piwikpro.github.io/ConsentManager-CustomConsentFormExample/
+
```
Comment (musialp) [inline consent_manager/js_api/reference.rst:2]:
```suggestion
```

Comment (archi-tektur) [reply consent_manager/js_api/reference.rst]: done

**Review by musialp — APPROVED** [review]

**Comment by archi-tektur** [comment]
@RomanSek @kororokke team review is done, please check it out and review if you have a moment

**Review by kororokke — APPROVED** [review]

**Pull request merged**